python_modules/dagster/dagster/_core/definitions/freshness_checks/last_update.py

```diff
@@ -100,10 +135,169 @@ def build_last_update_freshness_checks(
     )
+
+
+def _build_freshness_multi_check(
+    asset_keys: Sequence[AssetKey],
+    deadline_cron: Optional[str],
+    timezone: str,
+    severity: AssetCheckSeverity,
+    lower_bound_delta: datetime.timedelta,
+) -> AssetChecksDefinition:
+    params_metadata: dict[str, Any] = {TIMEZONE_PARAM_KEY: timezone}
+    if deadline_cron:
+        params_metadata[DEADLINE_CRON_PARAM_KEY] = deadline_cron
+    if lower_bound_delta:
+        params_metadata[LOWER_BOUND_DELTA_PARAM_KEY] = lower_bound_delta.total_seconds()
+
+    @multi_asset_check(
+        specs=[
+            AssetCheckSpec(
+                "freshness_check",
+                asset=asset_key,
+                metadata={FRESHNESS_PARAMS_METADATA_KEY: params_metadata},
+                description="Evaluates freshness for targeted asset.",
+            )
+            for asset_key in asset_keys
+        ],
+        can_subset=True,
+        name=f"freshness_check_{unique_id_from_asset_keys(asset_keys)}",
+    )
+    def the_check(context: AssetCheckExecutionContext) -> Iterable[AssetCheckResult]:
+        for check_key in context.selected_asset_check_keys:
+            asset_key = check_key.asset_key
+            current_timestamp = pendulum.now("UTC").timestamp()
+
+            current_time_in_freshness_tz = pendulum.from_timestamp(current_timestamp, tz=timezone)
+            latest_completed_cron_tick = (
+                get_latest_completed_cron_tick(
+                    deadline_cron, current_time_in_freshness_tz, timezone
+                )
+                if deadline_cron
+                else None
+            )
+            deadline = check.inst_param(
+                latest_completed_cron_tick or current_time_in_freshness_tz,
+                "deadline",
+                datetime.datetime,
+            )
+
+            last_update_time_lower_bound = cast(datetime.datetime, deadline - lower_bound_delta)
+
+            latest_record = retrieve_latest_record(
+                instance=context.instance, asset_key=asset_key, partition_key=None
+            )
+            update_timestamp = get_last_updated_timestamp(latest_record, context)
+            passed = (
+                update_timestamp is not None
+                and update_timestamp >= last_update_time_lower_bound.timestamp()
+            )
+
+            metadata: Dict[str, MetadataValue] = {
+                FRESHNESS_PARAMS_METADATA_KEY: JsonMetadataValue(params_metadata),
+                LOWER_BOUND_TIMESTAMP_METADATA_KEY: TimestampMetadataValue(
+                    last_update_time_lower_bound.timestamp()
+                ),
+            }
+            if latest_completed_cron_tick:
+                metadata[LATEST_CRON_TICK_METADATA_KEY] = TimestampMetadataValue(
+                    latest_completed_cron_tick.timestamp()
+                )
+            if passed:
+                # If the asset is fresh, we can potentially determine when it has the possibility of becoming stale again.
```

Comment:
Why only "potentially"?

Reply:
I think you're right that it's definitive.